File: app/src/main/java/com/denchic45/kts/data/model/room/TimestampConverter.java
package com.denchic45.kts.data.model.room;

import androidx.room.TypeConverter;

import java.util.Date;

public class TimestampConverter {

    @TypeConverter
    public long toLong(Date date) {
        if (date == null)
            return System.currentTimeMillis();
        return date.getTime();
    }

    @TypeConverter
    public Date toDate(long timestamp) {
        return new Date(timestamp);
    }

}

File: app/src/main/java/com/denchic45/kts/data/model/room/TimestampConverter.kt
package com.denchic45.kts.data.model.room

import androidx.room.TypeConverter
import java.util.*

class TimestampConverter {
    @TypeConverter
    fun toLong(date: Date?): Long? = date?.time

    @TypeConverter
    fun toDate(timestamp: Long?): Date? = timestamp?.let { Date(it) }

}